src/cli/commands/test/iac-local-execution/parsers/terraform-plan-parser.ts

```
@@ -47,23 +47,35 @@ function extractChildModulesResources(
 export function tryParsingTerraformPlan(
   terraformPlanFile: IacFileData,
 ): Array<IacFileParsed> {
-  // TODO: Handle failed parses with a meaningfull error message
-  const terraformPlanJson = JSON.parse(
-    terraformPlanFile.fileContent,
-  ) as TerraformPlanJson;
+  let terraformPlanJson;
+  try {
+    terraformPlanJson = JSON.parse(
+      terraformPlanFile.fileContent,
+    ) as TerraformPlanJson;
+  } catch (err) {
+    throw new Error('Failed to parse Terraform plan JSON file.');
+  }
+
+  try {
+    const rootModuleResources = extractRootModuleResources(terraformPlanJson);
+    const childModuleResources = extractChildModulesResources(
+      terraformPlanJson,
+    );
+    const parsedInput = [
+      ...rootModuleResources,
+      ...childModuleResources,
+    ].reduce(terraformPlanReducer, { resource: {}, data: {} });
 
-  // TODO: Handle missing fields in plan json with a meaningfull error message
-  const rootModuleResources = extractRootModuleResources(terraformPlanJson);
-  const childModuleResources = extractChildModulesResources(terraformPlanJson);
-  const parsedInput = [
-    ...rootModuleResources,
-    ...childModuleResources,
-  ].reduce(terraformPlanReducer, { resource: {}, data: {} });
-  return [
-    {
-      ...terraformPlanFile,
-      jsonContent: parsedInput,
-      engineType: EngineType.Terraform,
-    },
-  ];
+    return [
+      {
+        ...terraformPlanFile,
+        jsonContent: parsedInput,
+        engineType: EngineType.Terraform,
+      },
+    ];
+  } catch (err) {
+    throw new Error(
+      'Failed to extract resources from Terraform plan JSON file.',
+    );
```

Ditto here, we probably want to throw a specific Error subclass and detect here and then rethrow any other errors. Does the CLI have a logger we can run in debug mode to log the underlying error here to help users debug the issue?

---

agreed.
I want to include that work in the `CLI interface overhaul` ticket, to also map all of our potential errors and fine-tuning them with proper classes and messages.
not sure about the debug logger, worth checking.